### PR TITLE
[DC-972] Handle nested concepts for get participant count

### PR DESCRIPTION
**Pull request by evansuslovich**
Changed `CriteriaQueryBuilder` code:
- Change `c0.ancestor_concept_id = c.condition_concept_id to c0.descendant_concept_id = c.condition_concept_id`
- Remove `c.condition_concept_id = 'concept_id' OR`
- Created Singleton files in the `constants` package to represent field and column values from the tables:
   - `Concept`
   - `ConceptAncestor`
   - `ConceptRelationship`
   - `ConditionOccurrence`
   - `Observation`
   - `Person`
   - `ProcedureOccurrece`
 - Replace strings with constants

### CODE REVIEW

File: src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderConstants.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package bio.terra.service.snapshotbuilder.utils;
+
+public final class QueryBuilderConstants {
+
+  private QueryBuilderConstants() {}
```
Comment (evansuslovich) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderConstants.java]: I followed this article https://stackoverflow.com/questions/479565/how-do-you-define-a-class-of-constants-in-java and made a singleton file, thoughts on using an `interface`?

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java [hunk 2]
```diff
@@ -101,31 +104,25 @@ FilterVariable generateFilter(SnapshotBuilderDomainCriteria domainCriteria) {
     TableVariable ancestorVariable =
```
Comment (rjohanek) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java:103]: It could be nice to add some more java doc to this code!

Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java:103]: I completely agree!

File: src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java [hunk 3]
```diff
@@ -1,5 +1,10 @@
 package bio.terra.service.snapshotbuilder;
 
+import static bio.terra.service.snapshotbuilder.utils.QueryBuilderConstants.CONCEPT_ID;
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java]: In general, static imports are bad as they make it harder to see where a name is coming from. We make an exception for the mock and hamcrest APIs used in tests, but other cases shouldn't use them.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderConstants.java (hunk 1 above)
Comment (pshapiro4broad) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderConstants.java]: I would split these constants into two groups:

- `SELECT` aliases (or column names) which must be matched by result readers after a query is executed
- tables and their fields, used internally by query code

For aliases, those are part of the external contract of the query code, so I would put them in `QueryBuilderFactory` which is the API layer between the service code and the query code generation. If we wanted to close things up a bit more, we avoid using these constants outside `QueryBuilderFactory` and require callers to call methods like `getCount(ResultSet rs)` which would hide the way column names are represented. These methods would be on the query builder that generates the query.

For tables and fields, I think we should try putting each table and its columns in a single class, something like
```
class ConceptTable {
   public static final String NAME = "concept";
   public static final String CONCEPT_NAME = "concept_name";
   public static final String CONCEPT_CODE = "concept_code";
...
}
```

Also regarding select aliases vs column names, we may want to change our queries to always use aliases instead of relying on matching column names in the result, which would further isolate the aliases from the column names in queries.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java [hunk 4]
```diff
@@ -101,31 +125,25 @@ FilterVariable generateFilter(SnapshotBuilderDomainCriteria domainCriteria) {
...
             List.of(occurrenceVariable, ancestorVariable),
             new BooleanAndOrFilterVariable(
                 BooleanAndOrFilterVariable.LogicalOperator.OR,
                 List.of(
                     new BinaryFilterVariable(
                         new FieldVariable(
-                            new FieldPointer(occurrencePointer, domainOption.getColumnName()),
-                            occurrenceVariable),
-                        BinaryFilterVariable.BinaryOperator.EQUALS,
-                        new Literal(domainCriteria.getConceptId())),
```
Comment (rjohanek) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java:125]: Just curious, can you explain why we need this change?

File: src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderConstants.java (hunk 1 above)
Comment (rjohanek) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderConstants.java]: In future refactoring, would we want to move functions into this class as well? Like functions that represent shared query building code

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java (hunk 4 above)
Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java:125]: We don't need this because every concept is its own parent

File: src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java [hunk 5]
```diff
@@ -243,24 +243,26 @@ void generateRollupCountsQueryForCriteriaGroupsList(SqlRenderContext context) {
                             .meetAll(true)
                             .mustMeet(true))));
     // FIXME: is query correct? It doesn't contain the concept IDs 11 and 10.
```
Comment (rjohanek) [inline src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java:248]: is this the fix to this problem? or is there a separate ticket to address this problem? I still don't see concept ids 10 or 11 in the query

Comment (evansuslovich) [reply src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java:248]: This comment wasn't included in this PR, I'll give it a look and see if it's a ticket.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConditionOccurrenceConstants.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package bio.terra.service.snapshotbuilder.utils.constants;
+
+public final class ConditionOccurrenceConstants {
+  private ConditionOccurrenceConstants() { }
```
Comment (evansuslovich) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConditionOccurrenceConstants.java]: Thoughts on changing this to `OccurrenceConstants` and including other occurrences like `observation` with `observation_occurrence`, `observation_concept_id` and `procedure`, `procedure_occurrence`, and `procedure_concept_id`.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/PersonConstants.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package bio.terra.service.snapshotbuilder.utils.constants;
+
+public final class PersonConstants {
+  private PersonConstants() { }
+  public static final String PERSON = "person";
+  public static final String PERSON_ID = "person_id";
```
Comment (evansuslovich) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/constants/PersonConstants.java]: This could be in either `PersonConstants` or `ConditionOccurrenceConstants`? This value is present in both tables.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java [hunk 8]
```diff
@@ -43,7 +44,12 @@ private TablePointer getRootTablePointer() {
   private FieldVariable getFieldVariableForRootTable(String columnName) {
     return new FieldVariable(new FieldPointer(getRootTablePointer(), columnName), rootTable);
   }
-
+  /**
+   * Generate a filter for a SnapshotBuilderProgramDataRangeCriteria
+   * <pre>{@code
+   * p.year_of_birth >= low_value AND p.year_of_birth <= high_value
+   * </pre>
+   */
```
Comment (rjohanek) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java]: I know I previously advocated for SQL comments, but per our team conversation yesterday, I think we might actually want to remove these and instead make a test that shows the generated SQL.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderFactory.java [hunk 9]
```diff
@@ -5,6 +5,11 @@
 
 @Component
 public class QueryBuilderFactory {
+
+  public static final String HAS_CHILDREN = "has_children";
```
Comment (evansuslovich) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderFactory.java:9]: Putting the aliases inside `QueryBuilderFactory` was a suggestion, thoughts on making a new file for these?

Comment (rjohanek) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderFactory.java:9]: I don't have a strong opinion on whether they live here or in some alias file

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/PersonConstants.java (hunk 7 above)
Comment (rjohanek) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/constants/PersonConstants.java]: I'm not sure, what are your thoughts on putting it in both? I don't know if I even like that idea... or maybe you could put it in the QueryBuilderFactory since its shared...?

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java (hunk 8 above)
Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java]: They're removed!

File: src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java [hunk 10]
```diff
@@ -113,7 +115,12 @@ private static SnapshotBuilderProgramDataListOption generateSnapshotBuilderProgr
           .programDataOptions(
               List.of(
                   generateSnapshotBuilderProgramDataRangeOption(
-                      1, "person", "year_of_birth", "Year of birth", 0, 100),
+                      1,
+                      PersonConstants.PERSON,
+                      PersonConstants.YEAR_OF_BIRTH,
+                      "Year of birth",
```
Comment (rjohanek) [inline src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java]: Maybe this could be an alias constant too?

File: src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java [hunk 11]
```diff
@@ -113,7 +115,12 @@ private static SnapshotBuilderProgramDataListOption generateSnapshotBuilderProgr
...
+                      0,
+                      100),
                   generateSnapshotBuilderProgramDataListOption(
                       2,
                       "person",
```
Comment (rjohanek) [inline src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java]: Maybe this and these person columns below could be constants too?

Comment (evansuslovich) [reply src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java]: Good catch!

File: src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java (hunk 5 above)
Comment (rjohanek) [reply src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java:248]: Okay, I don't see one so I will file a new bug!

File: src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java (hunk 10 above)
Comment (rjohanek) [reply src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java]: And race, ethnicity, gender_identity maybe

Comment (evansuslovich) [reply src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java]: I'm going to change these tests because the column isn't `ethnicity` but `ethnicity_concept_id`

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java (hunk 8 above)
Comment (rjohanek) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java]: Okay, great and there tests that still represent the same info already?

Comment (rjohanek) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java]: Okay great and there are already tests that represent that same info, right?

**Review by rjohanek — APPROVED** [review]
I think this looks good, just minor comments on the refactor. Also, were you able to run the generated query in GCP console and get back real counts?

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConditionOccurrenceConstants.java (hunk 6 above)
Comment (rjohanek) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConditionOccurrenceConstants.java]: I kinda like the separateness of a class per table

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java [hunk 12]
```diff
@@ -0,0 +1,9 @@
+package bio.terra.service.snapshotbuilder.utils.constants;
+
+public final class ConceptAncestorConstants {
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java]: Could this be called `ConceptAncestor` instead? Does it need to contain `Constants`?

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+  private ConceptAncestorConstants() {}
+
+  public static final String CONCEPT_ANCESTOR = "concept_ancestor";
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java]: Since this field is the name of the table, and this class is the table, I would call this field `NAME`
```suggestion
  public static final String NAME = "concept_ancestor";
```

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/PersonConstants.java (hunk 7 above)
Comment (pshapiro4broad) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/constants/PersonConstants.java]: I would put it in both, the names are the same but they're different fields. At some later point I think these fields will be `private`

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java (hunk 8 above)
Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java]: Ya, I honestly made the comments from the tests!

File: src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java [hunk 14]
```diff
@@ -99,17 +95,21 @@ public Query buildConceptChildrenQuery(
   Query createSubQuery(int conceptId) {
     // concept_relationship is primary table for the subquery
     TableVariable conceptRelationship =
-        TableVariable.forPrimary(TablePointer.fromTableName(CONCEPT_RELATIONSHIP));
-    FieldVariable descendantConceptId = conceptRelationship.makeFieldVariable(CONCEPT_ID_2);
+        TableVariable.forPrimary(
+            TablePointer.fromTableName(ConceptRelationshipConstants.CONCEPT_RELATIONSHIP));
```
Comment (s-rubenstein) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java]: TOL: I wonder if `TABLE_NAME` might be a better variable name here? Or `NAME`? Then we could make the file `ConceptRelationshipTable` or `ConceptRelationship` or something - that way it can be more descriptive what it actually contains? What are your thoughts there?

If we wanted to impose more structure, you could add an object called `COLUMNS` that contained the column names, but that might be overly wordy and overkill.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java [hunk 15]
```diff
@@ -45,17 +35,21 @@ public Query buildConceptChildrenQuery(
       SnapshotBuilderDomainOption domainOption, int parentConceptId) {
 
     // concept table and its fields concept_name and concept_id
-    TableVariable concept = TableVariable.forPrimary(TablePointer.fromTableName(CONCEPT));
-    FieldVariable conceptName = concept.makeFieldVariable(CONCEPT_NAME);
-    FieldVariable conceptId = concept.makeFieldVariable(CONCEPT_ID);
+    TableVariable concept =
+        TableVariable.forPrimary(TablePointer.fromTableName(ConceptConstants.CONCEPT));
+    FieldVariable conceptName = concept.makeFieldVariable(ConceptConstants.CONCEPT_NAME);
+    FieldVariable conceptId = concept.makeFieldVariable(ConceptConstants.CONCEPT_ID);
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java]: This is OK for now, I don't think this PR is a great place to do any more than this level of refactoring. In fact, it might make sense to split out the string refactoring into its own PR so it's more apparent which changes are specific to the fix for DC-972.

But to give an idea about where I think this should go, I think we should have APIs more like:
```suggestion
    TableVariable concept = new Concept();
    FieldVariable conceptName = concept.name();
    FieldVariable conceptId = concept.id();
```

**Review by pshapiro4broad — COMMENTED** [review]
Some comments so far

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java [hunk 16]
```diff
@@ -97,35 +98,32 @@ FilterVariable generateFilter(SnapshotBuilderDomainCriteria domainCriteria) {
     TablePointer occurrencePointer = TablePointer.fromTableName(domainOption.getTableName());
     TableVariable occurrenceVariable = TableVariable.forPrimary(occurrencePointer);
 
-    TablePointer ancestorPointer = TablePointer.fromTableName("concept_ancestor");
+    TablePointer ancestorPointer =
+        TablePointer.fromTableName(ConceptAncestorConstants.CONCEPT_ANCESTOR);
```
Comment (s-rubenstein) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java]: Can we inline the pointer?

File: src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderFactory.java (hunk 9 above)
Comment (s-rubenstein) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderFactory.java:9]: This seems fine.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/PersonConstants.java (hunk 7 above)
Comment (s-rubenstein) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/constants/PersonConstants.java]: I wouldn't put it in the factory. My feeling is to put it in both, since technically there is nothing dictating that the two column names need to be the same. It's just an implementation detail of OMOP

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java [hunk 17]
```diff
@@ -97,35 +98,32 @@ FilterVariable generateFilter(SnapshotBuilderDomainCriteria domainCriteria) {
...
-            "ancestor_concept_id",
+            ConceptAncestorConstants.DESCENDANT_CONCEPT_ID,
             new FieldVariable(
                 new FieldPointer(occurrencePointer, domainOption.getColumnName()),
                 occurrenceVariable));
 
     return new SubQueryFilterVariable(
-        getFieldVariableForRootTable(PERSON_ID_FIELD_NAME),
+        getFieldVariableForRootTable(PersonConstants.PERSON_ID),
         SubQueryFilterVariable.Operator.IN,
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java]: While changing this, you could also update it to use `SubQueryFilterVariable.in()`, `BooleanAndOrFilterVariable.or()`, `BinaryFilterVariable.equals()`.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java (hunk 12 above)
Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java]: I suppose you don't need the `Constants` this is something that I thought about, I personally think it would read nicely to see the `Constants` when getting a term, to know that the variable is a constant. But since the value is in all capital letters it can already be assumed that the value is a constant.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java (hunk 15 above)
Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java]: Yes, I would love this, this is so much cleaner!!!

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java (hunk 13 above)
Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java]: I don't have a strong opinion on this!

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java (hunk 4 above)
Comment (pshapiro4broad) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java:125]: Is this change correct? Before it was
```
(occurrence.concept_id = domain_concept_id OR
 concept_ancestor.ancestor_concept_id = domain_concept_id)
```
and now it's
```
(concept_ancestor.ancestor_concept_id = domain_concept_id)
```

Maybe I'm misreading this but it looks like the occurrence table is no longer referenced in the WHERE clause. If so, then the SELECT will select all rows in the occurrence table.

Can you provide the entire SQL for this function (`generateFilter(SnapshotBuilderDomainCriteria)`) before and after this change?

File: src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderFactory.java (hunk 9 above)
Comment (pshapiro4broad) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/QueryBuilderFactory.java:9]: I think this is a good enough place for now.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java [hunk 18]
```diff
@@ -0,0 +1,9 @@
+package bio.terra.service.snapshotbuilder.utils.constants;
+
+public final class ConceptAncestorConstants {
+  private ConceptAncestorConstants() {}
```
Comment (s-rubenstein) [inline src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java]: You shouldn't need the constructor here, I wouldn't think? Java should take care of it for you.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java (hunk 13 above)
Comment (s-rubenstein) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java]: Seconding. Or `TABLE_NAME`

File: src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java (hunk 15 above)
Comment (s-rubenstein) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java]: Very much agreed - that seems like a great place to go with this.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java (hunk 12 above)
Comment (s-rubenstein) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java]: Yeah, I think the capital letters are enough there.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java (hunk 18 above)
Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/constants/ConceptAncestorConstants.java]: Java unfortunately doesn't handle this and an `issue` arises if there isn't a private constructor.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java (hunk 14 above)
Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java]: I like this suggestion! I removed all the`Constants` from the files and using `TABLE_NAME`.

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java (hunk 16 above)
Comment (evansuslovich) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java]: `ancestorPointer` is used in two places, line `105` and `125` do we still want to inline this?

File: src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java (hunk 4 above)
Comment (pshapiro4broad) [reply src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java:125]: I see, I missed that the `forJoined` above is joining on the occurrence table

File: src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java [hunk 19]
```diff
@@ -243,24 +246,26 @@ void generateRollupCountsQueryForCriteriaGroupsList(SqlRenderContext context) {
...
+    String expectedSql =
+        """
+      SELECT COUNT(DISTINCT p.person_id)
```
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java:251]: Can you reformat this query so it has some indentation? You can either follow the example below or copy the text into an IntelliJ SQL console and reformat it there.

**Review by pshapiro4broad — APPROVED** [review]
Looks good

File: src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java (hunk 19 above)
Comment (evansuslovich) [reply src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java:251]: got it!

**Review by pshapiro4broad — APPROVED** [review]
Looks good!
